src/cgroups/common.rs

```diff
@@ -45,8 +45,10 @@ pub fn write_cgroup_file_str<P: AsRef<Path>>(path: P, data: &str) -> Result<()>
         .create(false)
         .write(true)
         .truncate(false)
-        .open(path)?
-        .write_all(data.as_bytes())?;
+        .open(path.as_ref())
+        .with_context(|| format!("failed to open {:?}", path.as_ref()))?
+        .write_all(data.as_bytes())
+        .with_context(|| format!("failed to write to {:?}", path.as_ref()))?;
```

Comment:
Sometimes the OS error is indicative of cgroups errors, for example if you write an invalid value to a cgroup you'll get an "Invalid Argument" error. So it might be nice not to loose that context entirely. Perhaps you can just print the underlying error out in the formatting?

Reply:
I just realized in reading the anyhow docs that this will still print out the original error! Always read the docs!